NangoHQ/nango
# fix(sync): off-load data deletion

## Describe your changes

Fixes NAN-432

When deleting a connection or an integration, only soft deletes connections/syncs, the rest will be deleted by the cron. Ideally we would have a queue to listen and not bother with the weirdness of this script, but we don't, so we try to find everything that should be deleted but that it's not deleted.
I'm not entirely confident about the performance, I put some low LIMIT to avoid any issues.

- Stop hard deleting data when deleting connections/provider/syncs
- Add a cron that will swipe the data after the fact by selecting sync_id in the related table that should not exists

## Test

- Create a sync and run it
- Check `data_records`  
  - 1️⃣ You should see your records
- Delete the connection or the provider
- Check `data_records`
  - 2️⃣ You should see your records
- Wait 20minutes (or change the timing of the cron :D)
- Check `data_records` 
  - 3️⃣ You should **NOT** see your records
